File: app/src/main/kotlin/com/gengqiquan/githubhelper/utils/ExitUtil.java
package com.gengqiquan.githubhelper.utils;

import android.app.Activity;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by 耿 on 2016/8/30.
 */
public class ExitUtil {

    public static List<Activity> activity_ = new ArrayList<Activity>();

    public static void addActivity(Activity activity) {
        activity_.add(activity);
    }

    public static void removeActivity(Activity activity) {
        activity_.remove(activity);
    }

    public static void exit() {
        List<Activity> copy = new ArrayList<Activity>(activity_);
        for (Activity activity : copy) {
            activity.finish();
        }
        System.exit(0);
    }

    public static void finish() {
        List<Activity> copy = new ArrayList<Activity>(activity_);
        for (Activity activity : copy) {
            activity.finish();
        }
    }
    public static void finish(Class activity) {
        Activity act_=null;
        for (Activity act : activity_) {
            if(act.getClass().getName().equals(activity.getName())){
                act_=act;
            }
        }
        if (act_!=null){
            activity_.remove(act_);
            act_.finish();
        }
    }

}

File: app/src/main/kotlin/com/gengqiquan/githubhelper/utils/ExitUtil.kt
package com.gengqiquan.githubhelper.utils

import android.app.Activity

import java.util.ArrayList

/**
 * Created by 耿 on 2016/8/30.
 */
object ExitUtil {

    var activity_: MutableList<Activity> = ArrayList()

    fun addActivity(activity: Activity) {
        activity_.add(activity)
    }

    fun removeActivity(activity: Activity) {
        activity_.remove(activity)
    }

    fun exit() {
        val copy = ArrayList(activity_)
        for (activity in copy) {
            activity.finish()
        }
        System.exit(0)
    }

    fun finish() {
        val copy = ArrayList(activity_)
        for (activity in copy) {
            activity.finish()
        }
    }

    fun finish(activity: Class<*>) {
        var act_: Activity? = null
        for (act in activity_) {
            if (act.javaClass.name == activity.name) {
                act_ = act
            }
        }
        if (act_ != null) {
            activity_.remove(act_)
            act_.finish()
        }
    }

}
